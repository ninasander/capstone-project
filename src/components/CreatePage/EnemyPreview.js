import React from 'react'
import styled from 'styled-components/macro'
// import PropTypes from 'prop-types'

// EnemyPreview.propTypes = {
//   enemyName: PropTypes.string.isRequired,
//   armorClass: PropTypes.number.isRequired,
//   HP: PropTypes.number.isRequired,
//   initiative: PropTypes.number.isRequired,
// }

export default function EnemyPreview({
  enemyName,
  armorClass,
  HP,
  initiative,
  creatureEntry,
  handleDelete,
  onEdit,
}) {
  return (
    <>
      <DeleteButtonStyled onClick={handleDeleteClick} id={creatureEntry._id}>
        âœ•
      </DeleteButtonStyled>
      <EnemyEntryStyled>
        <h1>{initiative}</h1>
        <h2>{enemyName}</h2>
        <p>
          HP: {HP}/{HP}
        </p>
        <p>AC: {armorClass}</p>
        <EditButtonStyled onClick={onEdit}>Edit</EditButtonStyled>
      </EnemyEntryStyled>
    </>
  )

  function handleDeleteClick() {
    handleDelete(creatureEntry._id)
  }
}

const DeleteButtonStyled = styled.button`
  color: red;
  margin: 0;
  padding: 0;
  background: none;
  border-style: none;
  font-size: 110%;
  position: relative;
  left: 285px;
  top: 35px;
  z-index: 100;
`
const EditButtonStyled = styled.button`
  color: green;
  background: lightgreen;
  border-style: none;
  grid-column: span 3;
  border-radius: 5px;
  padding: 5px 20px;
  margin: 0;
  width: fit-content;
  justify-self: center;
`

const EnemyEntryStyled = styled.div`
  font-size: 80%;
  margin: 0 20px;
  padding: 20px;
  display: grid;
  grid-gap: 5px 20px;
  justify-content: space-between;
  grid-template-columns: 1fr 2fr 2fr;
  background: hsla(208, 40%, 28%, 1);
  background: linear-gradient(
      180deg,
      hsla(208, 40%, 28%, 1) 0%,
      hsla(208, 40%, 23%, 1) 51%,
      hsla(207, 38%, 18%, 1) 100%
    )
    fixed no-repeat;
  color: white;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8, 15, 21, 0.3);
  h1 {
    grid-row: span 2;
    align-self: center;
  }
  h2 {
    grid-column: span 2;
    margin: 0;
    font-size: 140%;
    word-break: break-word;
  }
  p {
    margin: 0;
  }
`
