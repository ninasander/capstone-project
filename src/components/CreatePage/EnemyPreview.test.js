import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import EnemyPreview from '../CreatePage/EnemyPreview'
import 'jest-styled-components'

describe('EnemyPreview', () => {
  const creatureEntry = {
    enemyName: 'Enemy 1',
    HP: 50,
    armorClass: 17,
    initiative: 20,
    _id: 5,
  }

  it('displays the enemy name', () => {
    const { getByText } = render(
      <EnemyPreview
        handleDelete={jest.fn()}
        handleEdit={jest.fn()}
        creatureEntry={creatureEntry}
        key={creatureEntry._id}
      />
    )

    expect(getByText(creatureEntry.enemyName)).toBeInTheDocument()
  })
  it('displays the current and max HP of the enemy', () => {
    const HPEntry = 'HP: ' + creatureEntry.HP + '/' + creatureEntry.HP
    const { getByText } = render(
      <EnemyPreview
        handleDelete={jest.fn()}
        handleEdit={jest.fn()}
        creatureEntry={creatureEntry}
        key={creatureEntry._id}
      />
    )

    expect(getByText(HPEntry)).toBeInTheDocument()
  })
  it('displays the enemy AC', () => {
    const acEntry = 'AC: ' + creatureEntry.armorClass
    const { getByText } = render(
      <EnemyPreview
        handleDelete={jest.fn()}
        handleEdit={jest.fn()}
        creatureEntry={creatureEntry}
        key={creatureEntry._id}
      />
    )

    expect(getByText(acEntry)).toBeInTheDocument()
  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <EnemyPreview
        handleDelete={jest.fn()}
        handleEdit={jest.fn()}
        creatureEntry={creatureEntry}
        key={creatureEntry._id}
      />
    )
    expect(tree).toMatchSnapshot()
  })
})
