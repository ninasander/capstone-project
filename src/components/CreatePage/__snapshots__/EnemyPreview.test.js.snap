// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EnemyPreview renders correctly 1`] = `
Array [
  .c0 {
  color: red;
  margin: 0;
  padding: 0;
  background: none;
  border-style: none;
  font-size: 110%;
  position: relative;
  left: 285px;
  top: 35px;
  z-index: 100;
}

<button
    className="c0"
    id={5}
    onClick={[Function]}
  >
    âœ•
  </button>,
  .c1 {
  color: green;
  background: lightgreen;
  border-style: none;
  grid-column: span 3;
  border-radius: 5px;
  padding: 5px 20px;
  margin: 0;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  justify-self: center;
}

.c0 {
  font-size: 80%;
  margin: 0 20px;
  padding: 20px;
  display: grid;
  grid-gap: 5px 20px;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  grid-template-columns: 1fr 2fr 2fr;
  background: hsla(208,40%,28%,1);
  background: linear-gradient( 180deg,hsla(208,40%,28%,1) 0%,hsla(208,40%,23%,1) 51%,hsla(207,38%,18%,1) 100% ) fixed no-repeat;
  color: white;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8,15,21,0.3);
}

.c0 h1 {
  grid-row: span 2;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0 h2 {
  grid-column: span 2;
  margin: 0;
  font-size: 140%;
  word-break: break-word;
}

.c0 p {
  margin: 0;
}

<div
    className="c0"
  >
    <h1>
      20
    </h1>
    <h2>
      Enemy 1
    </h2>
    <p>
      HP: 
      50
      /
      50
    </p>
    <p>
      AC: 
      17
    </p>
    <button
      className="c1"
      onClick={[Function]}
    >
      Edit
    </button>
  </div>,
]
`;
